log: harmonize bitcoind logging

jonatack opened the pull request:
Harmonize the user-facing output of the  `bitcoind -daemon`, `bitcoin-cli help stop`, `bitcoin-cli stop`, and `bitcoind -version` commands to be consistent with each other as well as with the "Bitcoin Core is probably already running" messages, e.g. `git grep 'probably already running.")'`.

Before:

```
$ bitcoind -regtest -daemon
Bitcoin Core daemon starting

$ bitcoind -regtest -daemon
Error: Bitcoin Core is probably already running.

$ bitcoind -regtest -version
Bitcoin Core Daemon version v0.18.99.0-e653eeff76-dirty

$ bitcoin-cli -regtest help stop
stop

Stop Bitcoin server.

$ bitcoin-cli -regtest stop
Bitcoin server stopping
```
these five commands output:

"Bitcoin Core daemon"
"Bitcoin Core"
"Bitcoin Core Daemon"
"Bitcoin server"
"Bitcoin server"

After this commit, they are all "Bitcoin Core".

```
$ bitcoind -regtest -daemon
Bitcoin Core starting

$ bitcoind -regtest -daemon
Error: Bitcoin Core is probably already running.

$ bitcoind -regtest -version
Bitcoin Core version v0.18.99.0-e90478f43e-dirty

$ bitcoin-cli -regtest help stop
stop

Request a graceful shutdown of Bitcoin Core.

$ bitcoin-cli -regtest stop
Bitcoin Core stopping
```

jonatack committed:
log: harmonize bitcoind server logging

Harmonize the user-facing output of the `bitcoind -daemon`, `bitcoin-cli help stop`, `bitcoin-cli stop`, and `bitcoind -version` commands to be consistent with each other as well as with the "Bitcoin Core is probably already running" messages, e.g. `git grep 'probably already running.")'`.

MarcoFalke commented on src/rpc/server.cpp:
```diff
@@ -162,7 +162,7 @@ UniValue stop(const JSONRPCRequest& jsonRequest)
         throw std::runtime_error(
             RPCHelpMan{"stop",
-                "\nStop Bitcoin server.",
+                "\nStop the " PACKAGE_NAME " daemon.",
```
The gui is not a daemon, so this should be kept more general

laanwj replied:
I don't think changing this from server to daemon is an improvement to be honest.
(definitely not from a documentation perspective, "server" is something everyone understands while "daemon" is very much a UNIX word)

promag replied:
How about "Requests a graceful shutdown"?

fanquake commented:
Concept ACK

practicalswift commented:
utACK 93f22ce41a828443b6cbc3662745e2f1de25d1fa

promag commented on src/rpc/server.cpp:
```diff
@@ -173,7 +173,7 @@ UniValue stop(const JSONRPCRequest& jsonRequest)
         MilliSleep(jsonRequest.params[0].get_int());
     }
-    return "Bitcoin server stopping";
+    return PACKAGE_NAME " daemon stopping";
```
Just `Stopping...`?

jonatack commented:
Thanks everyone for reviewing. Agreed, updated, and rebased. Two options for harmonizing the logging seem to be the simplicity of "Bitcoin Core", and the more specific but longer "Bitcoin Core server". Tentatively proposing the first as it is short and understandable with the bonus of not changing the other messages that output just "Bitcoin Core" e.g. `git grep 'probably already running.")'`.

promag commented on src/rpc/server.cpp:
```diff
@@ -162,7 +162,7 @@ UniValue stop(const JSONRPCRequest& jsonRequest)
         throw std::runtime_error(
             RPCHelpMan{"stop",
-                "\nStop Bitcoin server.",
+                "\nRequest a graceful shutdown of " PACKAGE_NAME ".",
```
What's the point of having the package name here?

jonatack replied:
Harmonization, consistency.

MarcoFalke commented on src/bitcoind.cpp:
```diff
@@ -143,7 +143,7 @@ static bool AppInit(int argc, char* argv[])
+            tfm::format(std::cout, PACKAGE_NAME " starting\n");
 
             // Daemonize
             if (daemon(1, 0)) { // don't chdir (1), do close FDs (0)
```
Why are you removing `daemon` here?

jonatack replied:
> Why are you removing `daemon` here?

https://github.com/bitcoin/bitcoin/pull/16489#discussion_r309873325 combined with the goal of harmonizing the various messages, thus my explanation here https://github.com/bitcoin/bitcoin/pull/16489#issuecomment-517408355.

jonatack replied:
I updated the PR description to explain the motivation.

jonatack commented:
I updated the PR description to hopefully better explain the motivation.

practicalswift commented:
ACK e90478f43e7bf9726ba033fde4a2776f9d5a9af4 (read code which looks good)

ongygy2000 commented:
RADIAZA
    On Sunday, August 4, 2019, 10:56:06 AM GMT+3, DrahtBot <notifications@github.com> wrote:  
 
 
The following sections might be updated with supplementary metadata relevant to reviewers and maintainers.

Conflicts

Reviewers, this pull request conflicts with the following ones:
   
   - #16544 (rpc: Add rpcallowstop by emilengler)

If you consider this pull request important, please also help to review the conflicting pull requests. Ideally, start with the one that should be merged first.


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.

jonatack commented:
Updated the PR description by adding:

these five commands output:

"Bitcoin Core daemon"
"Bitcoin Core"
"Bitcoin Core Daemon"
"Bitcoin server"
"Bitcoin server"

After this commit, they are all "Bitcoin Core".

practicalswift commented:
ACK e90478f43e7bf9726ba033fde4a2776f9d5a9af4 -- diff looks correct

jonatack commented:
Can this be merged? It would be good to have this in for 0.19.0.

ariard reviewed (approved):
Tested ACK e90478f

fjahr commented:
utACK e90478f

michaelfolkson commented:
ACK e90478f43e7bf9726ba033fde4a2776f9d5a9af4. Tested command outputs and as described.

fanquake reviewed (approved):
ACK e90478f43e7bf9726ba033fde4a2776f9d5a9af4
